game/yodesk.ksy

```diff
@@ -0,0 +1,564 @@
+meta:
+  id: yodesk
+  application: "Star Wars: Yoda Stories"
+  file-extension: dta
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: le
+  encoding: ASCII
+seq:
+  - id: catalog
+    type: catalog_entry
+    repeat: until
+    repeat-until: _.type == "ENDF"
+types:
+  catalog_entry:
+    seq:
+      - id: type
+        type: str
+        size: 4
+      - id: size
+        type: u4
+        if: type != "VERS" and type != "ZONE"
+      - id: content
+        type:
+          switch-on: type
+          cases:
+             '"VERS"': version
+             '"STUP"': setup_image
+             '"CHAR"': characters
+             '"CAUX"': character_auxiliaries
+             '"CHWP"': character_weapons
+             '"PUZ2"': puzzles
+             '"SNDS"': sounds
+             '"TILE"': tiles
+             '"TNAM"': tile_names
+             '"ZONE"': zones
+             '"ENDF"': endf
+             _: unknown_catalog_entry
+  unknown_catalog_entry:
+    seq:
+      - id: data
+        size: _parent.size
+  version:
+    seq:
+      - id: version
+        type: u4
+  setup_image:
+    seq:
+      - id: pixels
+        size: _parent.size
+  sounds:
+    seq:
+      - id: count
+        type: s2
+      - id: sounds
+        type: prefixed_strz
+        repeat: expr
+        repeat-expr: -count
+  tile_names:
+    seq:
+      - id: names
+        type: tile_name
+        repeat: until
+        repeat-until:  _.tile_id == 0xFF_FF
+  tile_name:
+    seq:
+      - id: tile_id
+        type: u2
+      - id: name
+        type: strz
+        size: 0x18
+        if: tile_id != 0xFF_FF
+  tiles:
+    seq:
+      - id: tiles
+        type: tiles_entries
+        size: _parent.size
+  tiles_entries:
+    seq:
+      - id: tiles
+        type: tile
+        repeat: eos
+  tile:
+      seq:
+        - id: attributes
+          type: u4
+        - id: pixels
+          size: 32 * 32
+  action:
+    seq:
+      - id: marker
+        contents: "IACT"
+      - id: size
+        type: u4
+      - id: condition_count
```

Comment:
`num_conditions` (https://doc.kaitai.io/ksy_style_guide.html#attr-id):

> For fields that designate **number / count** of something (in particular, number of repetitions of some other structure), use `num_` prefix and plural form — i.e. `num_questions`, `num_blocks`, `num_nodes`

Reply:
Thanks again for your feedback. I replaced all `_count` suffixes with `num_` prefixes.